platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -611,6 +621,12 @@ export default class HangingProtocolService extends PubSubService {
       return defaultReturn;
     }
 
+    const originalProtocol = this.protocolsCopy.get(this.protocol.id);
+    let originalProtocolStage;
+    if (!(originalProtocol instanceof Function)) {
```

Comment:
If the copy is done at the time the protocol is assigned, with the original/unaltered version, then it helps startup time as well as avoiding any of these instanceof functions - because the function will have already been called correctly.